cluster-autoscaler/cloudprovider/aws/instance_type_cache.go

```diff
@@ -0,0 +1,158 @@
+package aws
+
+import (
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"k8s.io/apimachinery/pkg/util/clock"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/client-go/tools/cache"
+	klog "k8s.io/klog/v2"
+)
+
+const (
+	asgInstanceTypeCacheTTL = time.Minute * 20
+	cacheMinTTL             = 120
+	cacheMaxTTL             = 600
```

Comment:
(only taking note for a potential follow-up, as this was already there (in an other file)):
perhaps asgInstanceTypeCacheTTL would deserve a cli arg so users can balance between throttling risks vs. freshness (depending on their context). And cacheMinTTL could be named something less confusing, like minCacheTTLJitter (resp. maxCacheTTLJitter).